src/js/background/sync.js

```diff
@@ -0,0 +1,461 @@
+      //   const storage = await sync.storageArea.get();
+      //   console.log("inSync: ", storage);
+      //   const localStorage = await browser.storage.local.get();
+      //   console.log("inLocal:", localStorage);
+      //   console.log("idents: ", await browser.contextualIdentities.query({}));
+      // }
+      console.log("Backed up!");
+      await sync.checkForListenersMaybeAdd();
+
+      async function updateSyncIdentities() {
+        const identities = await browser.contextualIdentities.query({});
+        await sync.storageArea.set({ identities });
+      }
+
+      async function updateCookieStoreIdMap() {
+        const cookieStoreIDmap = 
+          await identityState.getCookieStoreIDuuidMap();
+        await sync.storageArea.set({ cookieStoreIDmap });
+      }
+
+      async function updateSyncSiteAssignments() {
+        const assignedSites = 
+          await assignManager.storageArea.getAssignedSites();
+        await sync.storageArea.set({ assignedSites });
+      }
+
+      async function updateDeletedIdentityList(deletedIdentityUUID) {
+        const deletedIdentityList = 
+          await sync.storageArea.getDeletedIdentityList();
+        if (
+          deletedIdentityList.find(element => element === deletedIdentityUUID)
+        ) return;
+        deletedIdentityList.push(deletedIdentityUUID);
+        await sync.storageArea.set({ deletedIdentityList });
+      }
+
+      async function addToDeletedSitesList(siteStoreKey) {
+        const deletedSiteList = 
+          await sync.storageArea.getDeletedSiteList();
+        if (deletedSiteList.find(element => element === siteStoreKey)) return;
+        deletedSiteList.push(siteStoreKey);
+        await sync.storageArea.set({ deletedSiteList });
+      }
+
+      async function removeFromDeletedSitesList(siteStoreKey) {
+        const deletedSiteList = 
+          await sync.storageArea.getDeletedSiteList();
+        const newDeletedSiteList = deletedSiteList
+          .filter(element => element !== siteStoreKey);
+        await sync.storageArea.set({ deletedSiteList: newDeletedSiteList });
+      }
+    },
+
+    onChangedListener(changes, areaName) {
+      if (areaName === "sync") sync.errorHandledRunSync();
+    },
+
+    async addToDeletedList(changeInfo) {
+      const identity = changeInfo.contextualIdentity;
+      const deletedUUID = 
+        await identityState.lookupMACaddonUUID(identity.cookieStoreId);
+      await identityState.storageArea.remove(identity.cookieStoreId);
+      sync.storageArea.backup({uuid: deletedUUID});
+    },
+
+    async dataIsReliable() {
+      const cookieStoreIDmap = await this.getCookieStoreIDMap();
+      const identities = await this.getIdentities();
+      for (const cookieStoreId of Object.keys(cookieStoreIDmap)) {
+        const match = identities.find(identity => 
+          identity.cookieStoreId === cookieStoreId
+        );
+        // if one has no match, this is bad data.
+        if (!match) return false;
```

Comment:
```suggestion
        if (!match) {
          return false;
        }
```
Shorthand if statements aren't used elsewhere (or at least weren't last time I checked)

Reply:
Removed when data was restructured